### DISCUSSION TITLE
Cannot run k3s command via sudo command after install. *sudo: k3s: command not found.*

### DISCUSSION THREAD

**Discussion by hikarisakira** [k3s-io/k3s#7246] [Q&A] [answered]
**Environmental Info:**
K3s Version: 

```
k3s version v1.26.3+k3s1 (01ea3ff2)
go version go1.19.7
```

Node(s) CPU architecture, OS, and Version: 

`Linux hc-k8stest-01 5.14.0-162.22.2.el9_1.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Mar 15 14:44:24 EDT 2023 x86_64 x86_64 x86_64 GNU/Linux`

Cluster Configuration:

Haven't run any server or agents yet, just installed.

**Describe the bug:**

After installed k3s by running `curl -sfL https://get.k3s.io | sh -`, I was trying to run the server up, and runs k3s command by sudo, then sudo command says command not found.

**Steps To Reproduce:**

- Installed K3s: `curl -sfL https://get.k3s.io | sh -`
- Installer logs:
```
[INFO]  Finding release for channel stable
[INFO]  Using v1.26.3+k3s1 as release
[INFO]  Downloading hash https://github.com/k3s-io/k3s/releases/download/v1.26.3+k3s1/sha256sum-amd64.txt
[INFO]  Downloading binary https://github.com/k3s-io/k3s/releases/download/v1.26.3+k3s1/k3s
[INFO]  Verifying binary download
[INFO]  Installing k3s to /usr/local/bin/k3s
*dnf logs, on RHEL9*
正在更新訂閱管理程式軟體庫。
Rancher K3s Common (stable)   3.3 kB/s | 2.9 kB     00:00
依賴關係解析完畢。
==============================================================
 軟體包      架構   版本      軟體庫                     大小
==============================================================
安裝:
 k3s-selinux noarch 1.2-2.el8 rancher-k3s-common-stable  20 k

處理事項摘要
==============================================================
安裝  1 軟體包

總下載大小：20 k
安裝的大小：94 k
下載軟體包：
k3s-selinux-1.2-2.el8.noarch.  32 kB/s |  20 kB     00:00
--------------------------------------------------------------
總計                           32 kB/s |  20 kB     00:00
執行處理事項檢查
處理事項檢查成功。
執行處理事項測試
處理事項測試成功。
執行處理事項
  準備        :                                           1/1
  執行小令稿  : k3s-selinux-1.2-2.el8.noarch              1/1
  正在安裝    : k3s-selinux-1.2-2.el8.noarch              1/1
  執行小令稿  : k3s-selinux-1.2-2.el8.noarch              1/1
  核驗        : k3s-selinux-1.2-2.el8.noarch              1/1
安裝的產品已更新。

已安裝:
  k3s-selinux-1.2-2.el8.noarch

完成！
*dnf logs, on RHEL9*
[INFO]  Creating /usr/local/bin/kubectl symlink to k3s
[INFO]  Creating /usr/local/bin/crictl symlink to k3s
[INFO]  Creating /usr/local/bin/ctr symlink to k3s
[INFO]  Creating killall script /usr/local/bin/k3s-killall.sh
[INFO]  Creating uninstall script /usr/local/bin/k3s-uninstall.sh
[INFO]  env: Creating environment file /etc/systemd/system/k3s.service.env
[INFO]  systemd: Creating service file /etc/systemd/system/k3s.service
[INFO]  systemd: Enabling k3s unit
Created symlink /etc/systemd/system/multi-user.target.wants/k3s.service → /etc/systemd/system/k3s.service.
[INFO]  systemd: Starting k3s
```
- Runs k3s server by sudo command: `sudo k3s server`
`sudo: k3s: command not found.`

**Expected behavior:**

Should start k3s server up.

**Actual behavior:**

sudo command cannot find the k3s command after installed.

**Comment by brandond** [comment]
Make sure that /usr/local/bin is in your PATH. It sounds like that is not the case for your environment.

**Reply by hikarisakira** [reply]
I think that /usr/local/bin is in PATH now, I've been checked by running `echo $PATH`, and the result just like this:

`/usr/local/bin/k3s:/usr/local/bin:/home/Administrator/.g/bin:/home/Administrator/.g/go/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/Administrator/bin:/var/lib/snapd/snap/bin`

I also added them to my `~/.zshrc` though, by adding these following lines:
```
export PATH=/usr/local/bin:$PATH

export PATH=/usr/local/bin/k3s:$PATH
```
and nothing changes.

**Reply by brandond** [reply, marked as answer]
Sudo does not pass through environment variables. You should put it in the root path.

The quick start expects it to be in the default path for both root and unprivileged users, as it is on most distros.

**Reply by hikarisakira** [reply]
Oh, that helps!
After I edited via `sudo visudo`, it works.
Thanks for answering, and sorry for my stupid questions.😅
